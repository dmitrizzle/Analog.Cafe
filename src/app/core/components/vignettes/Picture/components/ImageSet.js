import LazyLoad from "react-lazyload"
import React from "react"

import { PicturePlaceholder } from "./PicturePlaceholder"
import { makeFroth } from "../../../../../utils"

export default props => {
  let src = props.src
  const className = props.className
  const classFeature = "Featured image"
  const classNofeature = "Supporting image"
  const by = " by "
  let alt
  if (props.alt) alt = props.alt
  else if (props.author && props.author.name)
    alt = props.feature
      ? classFeature + by + props.author.name
      : classNofeature + by + props.author.name
  else alt = props.feature ? classFeature : classNofeature
  let largestSize = props.feature ? "l" : "m"
  return (
    <PicturePlaceholder preserve frothId={src}>
      <picture>
        {!src.includes("data:image") &&
          makeFroth({ src, size: "s", type: "webp" }).type === "webp" && (
            <source
              srcSet={makeFroth({ src, size: "s", type: "webp" }).src}
              media="(max-width: 480px)"
              type="image/webp"
            />
          )}
        {!src.includes("data:image") &&
          makeFroth({ src, size: "s", type: "webp" }).type === "webp" && (
            <source
              srcSet={makeFroth({ src, size: "m", type: "webp" }).src}
              media="(max-width: 1200px)"
              type="image/webp"
            />
          )}
        {!src.includes("data:image") &&
          makeFroth({ src, size: "s", type: "webp" }).type === "webp" && (
            <source
              srcSet={makeFroth({ src, size: largestSize, type: "webp" }).src}
              media="(min-width: 1201px)"
              type="image/webp"
            />
          )}
        {!src.includes("data:image") && (
          <source
            srcSet={makeFroth({ src, size: "s" }).src}
            media="(max-width: 480px)"
          />
        )}
        {!src.includes("data:image") && (
          <source
            srcSet={makeFroth({ src, size: "m" }).src}
            media="(max-width: 1200px)"
          />
        )}
        {!src.includes("data:image") && (
          <source
            srcSet={makeFroth({ src, size: largestSize }).src}
            media="(min-width: 1201px)"
          />
        )}
        <LazyLoad unmountIfInvisible once offset={300} height={"100%"}>
          <img
            src={makeFroth({ src, size: "m" }).src}
            alt={alt}
            className={className}
            style={{ height: makeFroth({ src }).ratio ? "100%" : "initial" }}
          />
        </LazyLoad>
      </picture>
    </PicturePlaceholder>
  )
}
