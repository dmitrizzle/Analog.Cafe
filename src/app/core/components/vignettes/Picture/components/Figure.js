import React from "react"
import styled, { css } from "styled-components"

import { styles } from "../../Caption"
import ImageSet from "./ImageSet"
import Figcaption from "./Figcaption"

const bleed = css`
  float: none;
  margin-left: -${props => props.theme.size.block.padding}em;
  margin-right: -${props => props.theme.size.block.padding}em;
  margin-bottom: 0;
  width: 100vw !important;
  max-width: 100vw !important;
  box-shadow: none;
  border-radius: 0;
  ${props =>
    props.theme.size.breakpoint.min.l`margin-top: ${props =>
      props.theme.size.block.spacing}em;`} ${props =>
  props.feature
    ? props => props.theme.size.breakpoint.min.l`
		margin-left:	calc(( -100vw + ${props =>
      props.theme.size.block.column.m}px )/2 - ${props =>
        props.theme.size.block.padding}em );
	`
    : null} ${props =>
  props.feature
    ? props => props.theme.size.breakpoint.min.xxl`
		margin-left:	calc(( -100vw + ${props =>
      props.theme.size.block.column.l}px )/2 - ${props =>
        props.theme.size.block.padding}em );
	`
    : null} ${props => props.theme.size.breakpoint.max.m`
		margin-top: 0;
	`};
`
const shadow = css`
  box-shadow: 0 0 ${props => props.theme.size.block.spacing / 2}em
    ${props => props.theme.color.foreground(props.theme.opacity.least)};
`
const Figure = styled.figure`
  overflow: hidden;
  position: relative;
  padding: 0;
  margin: ${props => props.theme.size.block.spacing / 2}em
    ${props => props.theme.size.block.spacing}em
    ${props => props.theme.size.block.spacing}em -${props =>
  props.theme.size.block.column.m / 4}px;
  z-index: ${props => props.theme.layer.up};
  width: 85%;
  float: left;
  background: ${props => props.theme.color.background()};
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  ${shadow} ${props => props.theme.size.breakpoint.min.xxl`
		width: 				95%;
		margin-left: 	-${props => props.theme.size.block.column.l / 2.75}px;
		margin-right: ${props => props.theme.size.block.spacing}em;
	`} ${props =>
  !props.feature &&
  props.theme.size.breakpoint.max.l`
		float: none;
		margin: ${props => props.theme.size.block.spacing / 2}em 0 ${props =>
    props.theme.size.block.padding}em  -${props =>
    props.theme.size.block.padding}em !important;
		width: 75% !important;
		max-width: 66vw !important;
		min-width: ${props => props.theme.size.block.minFigureWIdth}px;
    &.focus {
      overflow: visible;
      &::after {
        content: "";
        width: 100vw;
        position: absolute;
        top: 0;
        bottom: 0;
        background:
            ${props =>
              props.theme.color.foreground(props.theme.opacity.least / 3)};
        z-index: ${props => props.theme.layer.tuck};
      }
    }
	`} ${props =>
  props.feature
    ? bleed
    : props => props.theme.size.breakpoint.max.m`
		margin-left: 0 !important;
		border-radius:	${props => props.theme.effects.borderRadius.small}em;
	`} ${props =>
  props.feature
    ? bleed
    : props => props.theme.size.breakpoint.max.s`
		${bleed}
		width: 100% !important;
		max-width: 100vw !important;
		min-width: 0;
		border-radius:	${props => props.theme.effects.borderRadius.small}em;
		${shadow}
	`} &.focus {
    border-top-right-radius: 0;
    border-top-left-radius: 0;
    box-shadow: 0 -${props => props.theme.size.block.border}px 0 ${props =>
  props.theme.color.highlight()};
    figcaption {
      box-shadow: 0 ${props => props.theme.size.block.border}px 0
        ${props => props.theme.color.highlight()} inset;
    }
    z-index: ${props => props.theme.layer.up + 1};
  }
  ${props =>
    props.feature &&
    !props.caption &&
    props.foldSpacer &&
    props.theme.size.breakpoint.min.l`
			margin-bottom: -${props.theme.size.block.spacing}em;
  `}
  textarea {
    ${styles};
    font-size: inherit !important;
  }
`

export default props => {
  const { src, ...select } = props
  return (
    <Figure {...select}>
      <ImageSet
        {...props}
        protected={
          props.readOnly !== false &&
          (process.env.NODE_ENV === "production" || props.userRole === "admin")
        }
      />
      <Figcaption
        caption={props.caption}
        nocaption={props.nocaption}
        readOnly={props.readOnly}
        focus={props.focus}
      >
        {props.children}
      </Figcaption>
    </Figure>
  )
}
