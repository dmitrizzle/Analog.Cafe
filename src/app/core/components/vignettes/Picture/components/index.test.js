import { shallow } from "enzyme"
import React from "react"

import Figcaption from "./Figcaption"
import Figure from "./Figure"
import ImageSet from "./ImageSet"
import Placeholder from "./Placeholder"

it("Render Figcaption without crashing", () => {
  shallow(<Figcaption />)
})

it("Render Figure without crashing", () => {
  shallow(<Figure />)
})

it("Render ImageSet without crashing", () => {
  shallow(<ImageSet />)
})

it("Render Placeholder without crashing", () => {
  shallow(<Placeholder />)
})
