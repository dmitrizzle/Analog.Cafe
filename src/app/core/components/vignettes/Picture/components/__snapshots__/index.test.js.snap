// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Figcaption, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Styled(styled.div) />,
      "style": Object {
        "borderBottom": "8px solid #2c2c2c",
        "height": 0,
        "overflow": "hidden",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "figcaption",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Styled(styled.div) />,
        "style": Object {
          "borderBottom": "8px solid #2c2c2c",
          "height": 0,
          "overflow": "hidden",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "figcaption",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render Figure, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Component)
          protected={false}
/>,
        undefined,
        <Unknown
          caption={undefined}
          captionInputFocus={undefined}
          focus={undefined}
          nocaption={undefined}
          readOnly={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "protected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "caption": undefined,
          "captionInputFocus": undefined,
          "children": undefined,
          "focus": undefined,
          "nocaption": undefined,
          "readOnly": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Component)
            protected={false}
/>,
          undefined,
          <Unknown
            caption={undefined}
            captionInputFocus={undefined}
            focus={undefined}
            nocaption={undefined}
            readOnly={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "protected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "caption": undefined,
            "captionInputFocus": undefined,
            "children": undefined,
            "focus": undefined,
            "nocaption": undefined,
            "readOnly": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render ImageSet, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(Component) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "className": "sc-bxivhb kpTgEk",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": "sc-bxivhb kpTgEk",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render Placeholder, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-htpNat NDSCX",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-htpNat NDSCX",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
